### ISSUE TITLE
Investigate the usage of a deviceController with unittest

### ISSUE THREAD

**Issue by zghp** [rdkcentral/python_raft#66]
## Goal:

- Explore the feasibility of transitioning device setup and management from `testController` to `deviceController`.

- Determine how unittest can be used to write and execute tests for the `deviceController`.

## Acceptance Criteria:

- Comprehensive understanding of the current `testController` implementation.

- Documented requirements and responsibilities for the `deviceController`.

- Clear outline of how `unittest` can be integrated with the `deviceController`.

- Proposed plan for implementing the `deviceController` and transitioning from `testController`.

**Comment by zghp** [comment]
Prepared an .md file with how this deviceController can be integrated, with pro/cons and other related notes

**Comment by zghp** [comment]
[ deviceManager.py - existing class, controls multiple device controllers ]
[ deviceController.py - contains everything needed to set up one device ]
 

- issue: we need our features + the unittest.TestCase features
   - solution: we need to run our main which then runs their main 

- issue: to get the rack config into deviceController (so it has everything it needs to set up the devices) we need to parse our arguments but to instantiate the class we have to call unittest main which has its own argument parser
    - solution: inherit their main, override it and then use it 


In short, we need to integrate a singleton so that the init of the unit test class and the init of the unit test main refer to it (this should work because unittest is still used as it was intended to but we override and set it up ourselves before we get there). And by calling super it means that both their parsers and our parsers are involved (thanks to the singleton)
